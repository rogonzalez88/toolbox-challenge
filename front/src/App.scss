@import "~bootstrap/scss/bootstrap";

#app {
  nav {
    background-color: #ff6666 !important;
    color: #fff;
    margin-bottom: 10px;
    .navbar-brand {
      color: #fff;
    }
  }
  thead {
    border-bottom: 2px solid #000;
  }
}