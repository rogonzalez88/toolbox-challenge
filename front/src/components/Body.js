import React, { useEffect } from 'react';

import { Table, Spinner } from 'react-bootstrap';
import { useSelector, useDispatch } from 'react-redux';

import { fetchFileData } from '../store/slices/filesSlice';

const Body = () => {
  const dispatch = useDispatch();
  const { tableData, isLoadingTableData } = useSelector(({ files }) => files);
  
  useEffect(() => {
    dispatch(fetchFileData());
  }, [dispatch]);

  if (isLoadingTableData)
    return (
      <div className="d-flex justify-content-center">
        <Spinner animation="border" />
      </div>
    );

  const tableBody = tableData
    .map((file) => file)
    .map(({ lines }) => lines)
    .flat();

  return (
    <Table striped bordered hover>
      <thead>
        <tr>
          {tableData.length > 0 && tableData[0].lines.length > 0 ? (
            Object.keys(tableData[0].lines[0]).map((header, index) => (
              <th className="text-capitalize" key={index}>
                {header}
              </th>
            ))
          ) : (
            <th>No Results</th>
          )}
        </tr>
      </thead>
      <tbody>
        {tableBody.map((content, index) => (
          <tr key={index}>
            {Object.values(content).map((value, valueIndex) => (
              <td key={valueIndex}>{value}</td>
            ))}
          </tr>
        ))}
      </tbody>
    </Table>
  );
};

export default Body;
