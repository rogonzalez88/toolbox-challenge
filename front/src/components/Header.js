import React, { useEffect } from 'react';

import { Container, Navbar, Form } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';

import { fetchFileList, fetchFileData } from '../store/slices/filesSlice';

const Header = () => {
  const dispatch = useDispatch();
  const { fileList, isLoadingFileList } = useSelector((state) => state.files);

  useEffect(() => {
    dispatch(fetchFileList());
  }, [dispatch]);

  const handleFilter = (newValue) => {
    if (newValue === 'all') {
      dispatch(fetchFileData());
    } else {
      dispatch(fetchFileData({ fileName: newValue }));
    }
    
  };

  return (
    <Navbar bg="light" expand="lg" fluid>
      <Container fluid>
        <Navbar.Brand>React Test App</Navbar.Brand>
        <Navbar.Toggle aria-controls="navbarScroll" />
        <Navbar.Collapse id="navbarScroll" className="justify-content-end">
          <Form.Group className="d-flex align-items-center w-25">
            <Form.Label className="w-50">Filter by </Form.Label>
            <Form.Select
              onChange={(event) => handleFilter(event.target.value)}
              disabled={isLoadingFileList}
              defaultValue="select"
            >
              <option value="select" disabled>
                Select file name
              </option>
              {fileList &&
                fileList.files.map((file, index) => (
                  <option value={file} key={index}>
                    {file}
                  </option>
                ))}
              <option value="all">
                All
              </option>
            </Form.Select>
          </Form.Group>
        </Navbar.Collapse>
      </Container>
    </Navbar>
  );
};

export default Header;
