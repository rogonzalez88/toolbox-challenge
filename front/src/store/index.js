import { configureStore } from "@reduxjs/toolkit";

import filesReducer from "./slices/filesSlice";

export default configureStore({
  reducer: {
    files: filesReducer,
  }
});
