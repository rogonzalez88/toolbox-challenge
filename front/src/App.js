import React from 'react';
import { Header, Table } from './components';
import { Container } from 'react-bootstrap';

const App = () => {
  return (
    <Container fluid>
      <Header />
      <Container>
        <Table />
      </Container>
    </Container>
  );
};

export default App;
