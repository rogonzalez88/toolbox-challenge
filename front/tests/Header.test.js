import React from 'react';

import renderer from "react-test-renderer";
import { Header } from "../src/components";
import WrapperReduxTest from "./utils/WrapperReduxTest";

it("Test Header component", () => {
  const component = renderer.create(
    <WrapperReduxTest
      initialState={{
        fileList: {
          files: [
            "test1.csv",
            "test2.csv",
            "test3.csv",
            "test4.csv",
            "test5.csv",
            "test6.csv",
            "test9.csv",
          ],
        },
        isLoadingFileList: false,
      }}
    >
      <Header />
    </WrapperReduxTest>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("Test Empty Header component", () => {
  const component = renderer.create(
    <WrapperReduxTest>
      <Header />
    </WrapperReduxTest>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
