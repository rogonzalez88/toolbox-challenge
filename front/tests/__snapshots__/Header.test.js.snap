// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Empty Header component 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
  fluid={true}
>
  <div
    className="container-fluid"
  >
    <span
      className="navbar-brand"
    >
      React Test App
    </span>
    <button
      aria-controls="navbarScroll"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="justify-content-end navbar-collapse collapse"
      id="navbarScroll"
    >
      <div
        className="d-flex align-items-center w-25"
      >
        <label
          className="w-50 form-label"
        >
          Filter by 
        </label>
        <select
          className="form-select"
          defaultValue="select"
          disabled={true}
          onChange={[Function]}
        >
          <option
            disabled={true}
            value="select"
          >
            Select file name
          </option>
          <option
            value="all"
          >
            All
          </option>
        </select>
      </div>
    </div>
  </div>
</nav>
`;

exports[`Test Header component 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
  fluid={true}
>
  <div
    className="container-fluid"
  >
    <span
      className="navbar-brand"
    >
      React Test App
    </span>
    <button
      aria-controls="navbarScroll"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="justify-content-end navbar-collapse collapse"
      id="navbarScroll"
    >
      <div
        className="d-flex align-items-center w-25"
      >
        <label
          className="w-50 form-label"
        >
          Filter by 
        </label>
        <select
          className="form-select"
          defaultValue="select"
          disabled={false}
          onChange={[Function]}
        >
          <option
            disabled={true}
            value="select"
          >
            Select file name
          </option>
          <option
            value="test1.csv"
          >
            test1.csv
          </option>
          <option
            value="test2.csv"
          >
            test2.csv
          </option>
          <option
            value="test3.csv"
          >
            test3.csv
          </option>
          <option
            value="test4.csv"
          >
            test4.csv
          </option>
          <option
            value="test5.csv"
          >
            test5.csv
          </option>
          <option
            value="test6.csv"
          >
            test6.csv
          </option>
          <option
            value="test9.csv"
          >
            test9.csv
          </option>
          <option
            value="all"
          >
            All
          </option>
        </select>
      </div>
    </div>
  </div>
</nav>
`;
